protocol-units/settlement/mcr/contracts/src/staking/MovementStaking.sol

```diff
@@ -70,69 +70,57 @@ contract MovementStaking is
             for (uint256 j = 0; j < custodiansByDomain[domain].length(); j++) {
                 address custodian = custodiansByDomain[domain].at(j);
 
-                // get the genesis stake for the attester
-                uint256 attesterStake = getStakeAtEpoch(
-                    domain,
-                    0,
-                    custodian,
-                    attester
-                );
-
-                // roll over the genesis stake to the current epoch
-                _addStake(
-                    domain,
-                    getCurrentEpoch(domain),
-                    custodian,
-                    attester,
-                    attesterStake
-                );
+                // for every delegatee of the attester
+                for (
+                    uint256 k = 0;
+                    k < delegatorsByAttesterByDomain[domain][attester].length();
```

Comment:
I don't know how good the optimizer is, but perhaps getting the `delegatorsByAttesterByDomain[domain][attester]` reference for the duration of the loop would save some gas.

Reply:
Yeah, I think it will actually save some gas.